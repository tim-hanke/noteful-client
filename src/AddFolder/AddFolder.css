.AddFolder {
  margin: 0;
  padding: 10px;
}

.AddFolder__form {
  --form-height: 100px;
  background-color: #010d1e;
  color: #cacaca;
  height: var(--form-height);
  margin-bottom: 8px;
  max-width: 380px;
  padding: 8px 0 8px 8px;
  position: relative;
}

.AddFolder__form label {
  display: block;
  font-family: var(--custom-font);
  margin: 0;
  max-width: 80%;
  min-height: 24px;
}

.AddFolder__form h2 {
  margin: 0;
  font-size: 20px;
}

.AddFolder__form input,
.AddFolder__form textarea,
.AddFolder__form select {
  font-size: 14px;
  padding: 8px;
  width: 300px;
}

.AddFolder__form select {
  height: 33px;
}

.AddFolder__form button {
  --form-button-height: 32px;
  background-color: #041a3a;
  border: 0;
  color: #bdbcbc;
  float: right;
  font-size: 12px;
  height: var(--form-button-height);
  position: relative;
  top: calc((var(--form-button-height) + 68px) * -1);
  transform-origin: right bottom;
  transform: rotate(-90deg);
  width: calc(var(--form-height));
}

.AddFolder__form button:hover,
.AddFolder__form button:focus {
  background-color: #29862e;
  color: white;
  cursor: pointer;
  font-weight: bold;
}

.AddFolder__form button:disabled {
  opacity: 0.3;
  background-color: #041a3a;
  color: #bdbcbc;
  font-weight: normal;
}

.required {
  color: #c51244;
}
