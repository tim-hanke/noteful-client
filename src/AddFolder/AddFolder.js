import React, { Component } from "react";
import { noteServer } from "../config";
import NotefulContext from "../NotefulContext/NotefulContext";
import ValidationError from "../ValidationError/ValidationError";
import "./AddFolder.css";

export default class AddFolder extends Component {
  constructor(props) {
    super(props);
    this.state = {
      folder: {
        name: "",
        touched: false,
      },
    };
  }

  static contextType = NotefulContext;

  updateFolderName(name) {
    this.setState({ folder: { name: name, touched: true } });
  }

  validateFolderName() {
    const name = this.state.folder.name.trim();
    if (name.length === 0) {
      return "(name is required)";
    }
  }

  async handleSubmit(e) {
    e.preventDefault();
    const { folder } = this.state;
    const options = {
      method: "POST",
      headers: { "content-type": "application/json" },
      body: JSON.stringify(folder),
    };

    try {
      const folderResponse = await fetch(noteServer + "/folders", options);

      if (!folderResponse.ok) {
        throw new Error("Something went wrong adding the folder.");
      }

      const folder = await folderResponse.json();

      this.context.addFolder(folder);
      this.props.history.goBack();
    } catch (error) {
      console.error(error);
    }
  }

  render() {
    const folderError = this.validateFolderName();
    console.log(folderError);
    return (
      <section className="AddFolder">
        <form
          className="AddFolder__form"
          onSubmit={(e) => {
            this.handleSubmit(e);
          }}
        >
          <label htmlFor="folderName">
            <h2>
              Folder Name:
              <span className="required"> * </span>
              {this.state.folder.touched && (
                <ValidationError message={folderError} />
              )}
            </h2>
          </label>
          <input
            type="text"
            id="folderName"
            name="folderName"
            aria-required="true"
            aria-invalid={folderError}
            value={this.state.folder.name}
            onChange={(e) => {
              this.updateFolderName(e.target.value);
            }}
          />
          <button type="submit" disabled={this.validateFolderName()}>
            Add Folder
          </button>
        </form>
      </section>
    );
  }
}
