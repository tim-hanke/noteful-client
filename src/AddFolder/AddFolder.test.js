import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AddFolder from "./AddFolder";

describe("AddFolder component", () => {
  it("renders an AddFolder componet by default", () => {
    const wrapper = shallow(<AddFolder />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
