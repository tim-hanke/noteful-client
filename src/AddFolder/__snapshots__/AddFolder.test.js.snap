// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFolder component renders an AddFolder componet by default 1`] = `
<section
  className="AddFolder"
>
  <form
    className="AddFolder__form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="folderName"
    >
      <h2>
        Folder Name:
        <span
          className="required"
        >
           * 
        </span>
      </h2>
    </label>
    <input
      aria-invalid="(name is required)"
      aria-required="true"
      id="folderName"
      name="folderName"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      disabled="(name is required)"
      type="submit"
    >
      Add Folder
    </button>
  </form>
</section>
`;
